### DISCUSSION TITLE
Unique behavior of jupyter in vscode

### DISCUSSION THREAD

**Discussion by xxc-zsz** [microsoft/vscode-jupyter#13881] [Questions and Answers] [answered]
The file structure is as follows

```
TESTJUPYTER
├─ .env
├─ test.ipynb
└─ test.py
```
Why can jupyter load the variables in the `.env` file without using the `python-dotenv` package?

![image](https://github.com/microsoft/vscode-jupyter/assets/108051416/a01a1fed-b015-4c72-a32d-283ff2d66ee3)

And it is only available in vscode. If I open it in jupyter on the web page, I cannot get the variable value.

![image](https://github.com/microsoft/vscode-jupyter/assets/108051416/bd5781c4-a75a-4ca2-bd9f-1a146f28dd9f)

So is there anything special about jupyter in vscode?

**Comment by xxc-zsz** [comment]
[Similar question in so.](https://stackoverflow.com/questions/76630107/vscodes-jupyter-notebook-environment-variables-mysteriously-loads-from-env)

**Reply by DonJayamanne** [reply, marked as answer]
Both Jupyter and Python extensions automatically load .env files in the following scenarios, there could be others as well
* running notebooks (jupyter extension)
* Running code in Interactive window (jupyter extension)
* Debugging Python code in python files (python extension)

**Reply by DonJayamanne** [reply]
Closing as answered
